### PR TITLE
Update and link ES exception handling

### LINKED ISSUE #414: "Handling exceptions" section in ES contains outdated information
https://heycam.github.io/webidl/#es-handling-exceptions says:

> None of the algorithms or processing requirements in the ECMAScript language binding catch ECMAScript exceptions.

But it is not so in [attribute getter](https://heycam.github.io/webidl/#dfn-attribute-getter)'s definition:

> ...
> And then, if an exception was thrown:
>
> 1. If op has a return type that is a promise type, then:
>     ...
> 2. Otherwise, end these steps and allow the exception to propagate.

### PR DESCRIPTION
**Pull request by TimothyGu**
Fixes #414.

***
[Preview](https://s3.amazonaws.com/pr-preview/TimothyGu/webidl/exception-handling-update.html) | [Diff](https://s3.amazonaws.com/pr-preview/heycam/webidl/48f1e62...TimothyGu:c6752b0.html)

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -13110,12 +13123,14 @@ The following typographic conventions are used in this document:
 The following conventions are used in the algorithms in this document:
 
 *   Algorithms use the [=conventions of the ECMAScript specification=],
-    including the ! and ? notation for unwrapping completion records.
-*   Algorithms sometimes treat returning/throwing values and returning completion records
+    including the ! and ? notation for unwrapping [=Completion Records=].
```
Comment (tobie) [inline index.bs]: Would have been nice to link directly to the Completion Record dfn, here, if you're so inclined to file a bug with the TC-39 spec.

File: index.bs [hunk 2]
```diff
@@ -13110,12 +13123,14 @@ The following typographic conventions are used in this document:
...
+*   Algorithms sometimes treat returning/throwing values and returning [=Completion Records=]
     interchangeably. That is, an algorithm that uses return/throw terminology may be treated as
-    returning a completion record, while one that returns a completion record may be treated as
-    returning a value or throwing an exception.
-*   Completion records are extended by allowing them to contain values that are not ECMAScript
+    returning a [=Completion Record=], while one that returns a [=Completion Record=] may be treated as
+    returning a value or throwing an exception. Similarly, to catch exceptions, defining the
+    behavior to adopt [=if an exception was thrown=] and checking if the [=Completion Record=]’s
+    \[[Type]] field is “throw” are equivalent as well.
```
Comment (tobie) [inline index.bs]: "as well" is a bit redundant, here.

File: index.bs [hunk 3]
```diff
@@ -12626,14 +12629,17 @@ A {{DOMException}} is represented by a
...
-ends due to an exception being thrown, that exception
+Unless specified otherwise,
+whenever ECMAScript runtime semantics are invoked due
+to requirements in this document and
+end due to an exception being thrown, that exception
 must propagate to the caller, and if
 not caught there, to its caller, and so on.
 
+Per [[#conventions]], an algorithm specified in this document may intercept thrown exceptions, either by specifying
+the exact steps to take <dfn id="dfn-if-an-exception-was-thrown">if an exception was thrown</dfn>, or by explicitly
```
Comment (tobie) [inline index.bs]: If feels weird to integrate the conditional ("if") within the DFN. I'd be more confortable with:

```
<dfn id="dfn-an-exception-was-thrown">an exception was thrown</dfn>
```

File: index.bs (hunk 1 above)
Comment (tobie) [inline index.bs]: Nice, but not a show stopper.

File: index.bs (hunk 3 above)
Comment (domenic) [inline index.bs]: I think the id is not necessary either; we include it a lot elsewhere to preserve pre-Bikeshed IDs, but we can use the autogenerated one when introducing new terms.

Comment (TimothyGu) [inline index.bs]: Ah didn't know the id could be autogenerated. Will fix.

File: index.bs (hunk 1 above)
Comment (TimothyGu) [inline index.bs]: The link does go to Completion Record directly: https://tc39.github.io/ecma262/#sec-completion-record-specification-type

File: index.bs [hunk 4]
```diff
@@ -75,10 +75,11 @@ urlPrefix: https://tc39.github.io/ecma262/; spec: ECMA-262
         text: abs; url: eqn-abs
         text: modulo; url: eqn-modulo
         url: sec-algorithm-conventions
-            text: !
-            text: ?
             text: ECMA-262 section 5.2
             text: conventions of the ECMAScript specification
+        url: sec-returnifabrupt-shorthands
+            text: !
+            text: ?
```
Comment (tobie) [inline index.bs]: Oh, being able to link directly to an explanation of what the shorthand does is amazing!

**Pull request merged**